src/lib/update-alternatives/model/alternative.rb

```diff
@@ -48,13 +48,15 @@ def self.all_names
 
     # @return [Array<Alternative>] an array with all alternatives.
     def self.all
-      all_names.map { |name| load(name) }
+      all_names.map { |name| load(name) }.compact
     end
 
     # @return [Alternative] an alternative with the given name.
     # @param name [String] The name of the alternative to be loaded.
     def self.load(name)
       raw_data = Cheetah.run("update-alternatives", "--query", name, stdout: :capture).lines
+      empty_line = raw_data.find_index("\n")
+      return nil if empty_line.nil?
```

Comment:
About the first point, that is an interesting design pattern, I will use it.

For the second point, I think it is a good indicator because the empty line is used in the output of `update-alternatives --query an-alternative-name` to separate the alternative data and the choices, and if there are more than one choice, is used to separate that choices.